src/renderer/components/commands-action-button.tsx

```diff
@@ -259,21 +259,15 @@ export class GistActionButton extends React.Component<
     const { gistId } = this.props.appState;
     const { actionType } = this.state;
 
-    if (gistId) {
-      switch (actionType) {
-        case GistActionType.publish:
-          await this.handlePublish();
-          break;
-        case GistActionType.update:
-          await this.handleUpdate();
-          break;
-        case GistActionType.delete:
-          await this.handleDelete();
-          break;
-      }
-    } else {
-      await this.handlePublish();
+    if (gistId && actionType === GistActionType.update) {
+      return this.handleUpdate();
+    }
+
+    if (gistId && actionType === GistActionType.delete) {
+      return this.handleDelete();
```

Comment:
this pattern is a little confusing since there's no return value to speak of here - what motivated this change?

Reply:
The main motivation for refactoring performGistAction() is to reduce the number of logic branches needed to reach the three possible outcomes of that function (publish, update, delete). However now that I reread it, I see three more branches can be removed while making the intent of the 'return' functions clearer. Revised in cff84ed.

The return values are the promises created by handlePublish(), handleDelete(), and handleDelete(). I don't feel super strongly about this, but my thinking was that by removing the unncessary await it avoids unnecessary microtasks.